app/scripts/directives/resources.js

```diff
@@ -1,6 +1,75 @@
 'use strict';
 
 angular.module('openshiftConsole')
+  .directive('containerStatuses', function($filter) {
+    return {
+      restrict: 'E',
+      scope: {
+        containerStatuses: '=',
+        startDebugTerminal: '=?',
+        detailed: '=?'
+      },
+      templateUrl: 'views/_container-statuses.html',
+      link: function(scope) {
+        scope.hasDebugTerminal = angular.isFunction(scope.startDebugTerminal);
+
+        scope.podContainerStatuses = _.get(scope.containerStatuses, 'status.containerStatuses', []);
+        scope.podContainerInitStatuses = _.get(scope.containerStatuses, 'status.initContainerStatuses', []);
+
+        scope.hasContainerFailed = function (containerStatus) {
+          return $filter('isContainerFailed')(containerStatus) !== false;
+        };
+
+        scope.haveContainersFailed = function(containerStatuses) {
+
+          var failed = false;
+
+          _.forEach(containerStatuses, function(containerStatus) {
+            if (scope.hasContainerFailed(containerStatus)) {
+              failed = true;
+              return false;
+            }
+          });
+
+          return true;
+        };
+
+        scope.expandInitContainers = scope.initContainersFailed = scope.haveContainersFailed(scope.podContainerInitStatuses);
```

Comment:
This won't update when the pod status changes. I'd expect the section to automatically collapse if the init containers complete when you're on the page.